src/Assets/Asset2D.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Asset2D.hpp"
+
+Asset2D::Asset2D()
+{
+    _textureLoaded = false;
+}
+
+Asset2D::~Asset2D()
+{
+}
+
+Texture2D Asset2D::getTexture()
+{
+    if (_textureLoaded)
```

Comment:
the _texture should exist anyway, she should be initialised in the constructor, so the getter don't need condition to return (also warning, if there if the condition isn't fullfiled, you don't return anything here)